### PR TITLE
[DO NOT MERGE] [base] Get back to boost regexp.

### PR DESCRIPTION
**Pull request by ygorshenin**
Ubuntu 14.04 has a serious bug in a libc++ which prevents us from usage of this library. This bug was fixed in a later versions, so, if our testing machine will be upgraded to modern versions of Ubuntu the bug will go away as libc++ will be updated. But, if this will not happen, the only possibility is to get back to Ubuntu 14.04 libstdc++, which does not support std::regex and other fancy features.

To sum up: in case of successful Jenkins linux bot upgrade this CL must be closed. Otherwise, it must be merged.

### CODE REVIEW

**Comment by ygorshenin** [comment]
Closing the PR as bots happy now.

**Comment by deathbaba** [comment]
What about libstdc++? Does it work with it?

**Comment by ygorshenin** [comment]
Could you please paraphrase your question? It's unclear what are you talking about.

**Comment by deathbaba** [comment]
Does your code compile on ubuntu with -spec linux-clang ?

**Comment by ygorshenin** [comment]
Sorry, man. I can't check our project on all ubuntus and all libstdc++'s, especially on those that don't support c++11.

**Comment by ygorshenin** [comment]
If you have enough resources for doing that, please, share results with us.

**Comment by deathbaba** [comment]
I don't understand your reply.
My question is simple: does it work on the same ubuntu 14.04 but with libstdc++ instead of libc++? Our code compiles well without libc++ (only recent update of opening_hours broke it).

**Comment by ygorshenin** [comment]
Our code does not compile with libstdc++s not supporting C++11. Our code compiles with libstdc++s supporting C++11.

**Comment by deathbaba** [comment]
Of course, libstdc++ supports c++11 too. Why do you point to C++11? I just asked if you tried with another std library, not about turning off C++11.

**Comment by ygorshenin** [comment]
Your question was " What about libstdc++? Does it work with it?", and I
gave you exact reply. If you think that I didn't, don't hesitate to reread
my previous messages instead of writing new ones.

**Comment by deathbaba** [comment]
> Ubuntu 14.04 has a serious bug in a libc++ which prevents us from usage of this library. This bug was fixed in a later versions, so, if our testing machine will be upgraded to modern versions of Ubuntu the bug will go away as libc++ will be updated. But, if this will not happen, the only possibility is to get back to Ubuntu 14.04 libstdc++, which does not support std::regex and other fancy features.

Му question was "have you tried to compile your code in exactly the same environment, but without using libc++, which has a bug, and using libstdc++ instead? With exactly the same C++11 options, which are not related at all to the standard library used (e.g. remove -stdlib=libc++ option from command line).
